src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -661,9 +661,15 @@ public function isPublic()
      * @return Definition The current instance
      *
      * @api
+     *
+     * @deprecated since version 2.7, will be removed in 3.0.
      */
-    public function setSynchronized($boolean)
+    public function setSynchronized($boolean, $triggerDeprecationError = true)
```

Comment:
thus, the Definition class is not really meant to be a class extended by inheritance. I don't see any use case where this would bring any benefit actually.